src/machines/continue_as_new_workflow_state_machine.rs

```diff
@@ -35,6 +56,110 @@ impl Created {
     pub(super) fn on_schedule(
         self,
     ) -> ContinueAsNewWorkflowMachineTransition<ContinueAsNewWorkflowCommandCreated> {
-        unimplemented!()
+        TransitionResult::default()
+    }
+}
+
+impl From<ContinueAsNewWorkflowCommandCreated> for ContinueAsNewWorkflowCommandRecorded {
+    fn from(_: ContinueAsNewWorkflowCommandCreated) -> Self {
+        Default::default()
+    }
+}
+
+impl TryFrom<HistoryEvent> for ContinueAsNewWorkflowMachineEvents {
+    type Error = WFMachinesError;
+
+    fn try_from(e: HistoryEvent) -> Result<Self, Self::Error> {
+        Ok(match EventType::from_i32(e.event_type) {
+            Some(EventType::WorkflowExecutionContinuedAsNew) => {
+                Self::WorkflowExecutionContinuedAsNew
+            }
+            _ => {
+                return Err(WFMachinesError::UnexpectedEvent(
+                    e,
+                    "Continue as new workflow machine does not handle this event",
+                ))
+            }
+        })
+    }
+}
+
+impl TryFrom<CommandType> for ContinueAsNewWorkflowMachineEvents {
+    type Error = ();
+
+    fn try_from(c: CommandType) -> Result<Self, Self::Error> {
+        Ok(match c {
+            CommandType::ContinueAsNewWorkflowExecution => {
+                Self::CommandContinueAsNewWorkflowExecution
+            }
+            _ => return Err(()),
+        })
+    }
+}
+
+impl WFMachinesAdapter for ContinueAsNewWorkflowMachine {
+    fn adapt_response(
+        &self,
+        _event: &HistoryEvent,
+        _has_next_event: bool,
+        _my_command: Self::Command,
+    ) -> Result<Vec<MachineResponse>, WFMachinesError> {
+        Ok(vec![])
+    }
+}
+
+impl Cancellable for ContinueAsNewWorkflowMachine {}
```

Comment:
We don't have any use cases for continue as new cancellation, right?

Reply:
No, hence why the implementation is empty. All machines are required to implement the trait but only the ones that override the method it defines do anything.